src/modelfree/common/utils.py

```diff
@@ -59,15 +60,23 @@ def __init__(self, policy):
         """
         super().__init__(policy=policy, sess=policy.sess)
 
-    def predict(self, observation, state=None, mask=None, deterministic=False):
+    def predict(self, observation, state=None, mask=None, deterministic=False, return_data=True):
         if state is None:
             state = self.policy.initial_state
         if mask is None:
             mask = [False for _ in range(self.policy.n_env)]
 
-        actions, _val, states, _neglogp = self.policy.step(observation, state, mask,
-                                                           deterministic=deterministic)
-        return actions, states
+        if isinstance(self.policy, TransparentPolicy):
+            actions, _val, states, _neglogp, data = self.policy.step(observation, state, mask,
+                                                                     deterministic=deterministic)
+            if return_data:
+                return actions, states, data
+            else:
+                return actions, states
+        else:
+            actions, _val, states, _neglogp = self.policy.step(observation, state, mask,
```

Comment:
I'm not sure if this would be preferable, but one possibility to make this logic a little cleaner is to just to run the policy in a way that captures the returns into a tuple, and then returns the data-aligned element in the tuple. For example, 

```
policy_step_return_tuple = self.policy.step(observation, state, mask, deterministic=deterministic) 
if isinstance(self.policy, TransparentPolicy) and return_data: 
    return policy_step_return_tuple[0], policy_step_return_tuple[2]
else: 
    return policy_step_return_tuple[0], policy_step_return_tuple[2], policy_step_return_tuple[4]
```
This has the advantage of not requiring as complicated conditional logic, but the downside of being less transparent because you don't explicitly variable-name what you're returning in each case, meaning you'd probably want to indicate that in comments